core/services/workflows/engine.go

```diff
@@ -680,6 +681,40 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 	}
 }
 
+func merge(baseConfig *values.Map, overrideConfig *values.Map) *values.Map {
+	m := values.EmptyMap()
+
+	for k, v := range baseConfig.Underlying {
+		m.Underlying[k] = v
+	}
+
+	for k, v := range overrideConfig.Underlying {
+		m.Underlying[k] = v
+	}
+
+	return m
+}
+
+func configForStep(ctx context.Context, reg core.CapabilitiesRegistry, localNode capabilities.Node, step *step) (*values.Map, error) {
+	ID := step.info.ID
+
+	// If the capability info is missing a DON, then
+	// the capability is local, and we should use the localNode's DON ID.
+	var donID uint32
+	if step.info.DON != nil && step.info.DON.ID > 0 {
+		donID = step.info.DON.ID
+	} else {
+		donID = localNode.WorkflowDON.ID
+	}
+
+	capConfig, err := reg.ConfigForCapability(ctx, ID, donID)
+	if err != nil {
+		return nil, err
+	}
+
+	return merge(capConfig.ExecuteConfig, step.config), nil
```

Comment:
Suggested rename to `capConfig.DefaultConfig`

Reply:
Add comment for "currently, all config for a DON can be overridden for a capability, which is probably not something we want..."